PowerPointLabs/PowerPointLabs/CropLab/CropOutPadding.cs

```diff
@@ -0,0 +1,224 @@
+            }
+
+            for (int i = 1; i <= shapeRange.Count; i++)
+            {
+                float currentRotation = shapeRange[i].Rotation;
+                float cropLeft = shapeRange[i].PictureFormat.CropLeft;
+                float cropRight = shapeRange[i].PictureFormat.CropRight;
+                float cropTop = shapeRange[i].PictureFormat.CropTop;
+                float cropBottom = shapeRange[i].PictureFormat.CropBottom;
+
+                shapeRange[i].PictureFormat.CropLeft = 0;
+                shapeRange[i].PictureFormat.CropRight = 0;
+                shapeRange[i].PictureFormat.CropTop = 0;
+                shapeRange[i].PictureFormat.CropBottom = 0;
+                shapeRange[i].Rotation = 0;
+
+                Utils.Graphics.ExportShape(shapeRange[i], TempPngFileExportPath);
+                using (Bitmap shapeImage = new Bitmap(TempPngFileExportPath))
+                {
+                    PowerPoint.ShapeRange origShape = shapeRange[i].Duplicate();
+                    origShape.ScaleWidth(1, Office.MsoTriState.msoTrue);
+                    origShape.ScaleHeight(1, Office.MsoTriState.msoTrue);
+                    float origWidth = origShape.Width;
+                    float origHeight = origShape.Height;
+                    origShape.Delete();
+
+                    Rectangle cropRect = GetImageBoundingRect(shapeImage);
+                    float cropRatioLeft = cropRect.Left / (float)shapeImage.Width;
+                    float cropRatioRight = (shapeImage.Width - cropRect.Width) / (float)shapeImage.Width;
+                    float cropRatioTop = cropRect.Top / (float)shapeImage.Height;
+                    float cropRatioBottom = (shapeImage.Height - cropRect.Height) / (float)shapeImage.Height;
+
+                    float newCropLeft = origWidth * cropRatioLeft;
+                    float newCropRight = origWidth * cropRatioRight;
+                    float newCropTop = origHeight * cropRatioTop;
+                    float newCropBottom = origHeight * cropRatioBottom;
+
+                    cropLeft = cropLeft < newCropLeft ? newCropLeft : cropLeft;
+                    cropRight = cropRight < newCropRight ? newCropRight : cropRight;
+                    cropTop = cropTop < newCropTop ? newCropTop : cropTop;
+                    cropBottom = cropBottom < newCropBottom ? newCropBottom : cropBottom;
+                }
+
+                shapeRange[i].Rotation = currentRotation;
+                shapeRange[i].PictureFormat.CropLeft = cropLeft;
+                shapeRange[i].PictureFormat.CropRight = cropRight;
+                shapeRange[i].PictureFormat.CropTop = cropTop;
+                shapeRange[i].PictureFormat.CropBottom = cropBottom;
+            }
+
+            return shapeRange;
+        }
+
+        private static bool IsImageRowTransparent(Bitmap image, int y)
+        {
+            for (int x = 0; x < image.Width; x++)
+            {
+                if (image.GetPixel(x, y).A > 0)
```

Comment:
Fixed, loaded bitmap into memory using lockbits which is much faster than GetPixels method.